### PR TITLE
SNOW-1486910: Return DatetimeIndex from to_datetime and date_range

### PR DESCRIPTION
**Pull request by sfc-gh-nkumar**
1. Which Jira issue is this PR addressing? Make sure that there is an accompanying issue to your PR.

   

   Fixes SNOW-1486910

2. Fill out the following pre-review checklist:

   - [ ] I am adding a new automated test(s) to verify correctness of my new code
      - [ ] If this test skips Local Testing mode, I'm requesting review from @snowflakedb/local-testing
   - [ ] I am adding new logging messages
   - [ ] I am adding a new telemetry message
   - [ ] I am adding new credentials
   - [ ] I am adding a new dependency
   - [ ] If this is a new feature/behavior, I'm adding the Local Testing parity changes.

3. Please describe how your code solves the related issue.

   Return DatetimeIndex from pd.to_datetime and pd.date_range methods.

### CODE REVIEW

**Review by sfc-gh-helmeleegy — COMMENTED** [review]
Looks good. Can you also update the relevant doctests under the `docstrings/` directory? Some doctests are currently skipped because they return DatetimeIndex.

**Comment by sfc-gh-nkumar** [comment]
> Looks good. Can you also update the relevant doctests under the `docstrings/` directory? Some doctests are currently skipped because they return DatetimeIndex.

Updated doctests. There are few still skipped which will be fixed in next PR (SNOW-1558888) which will add support for datetime properties.

**Review by sfc-gh-helmeleegy — APPROVED** [review]
LGTM. Thank!

File: src/snowflake/snowpark/modin/pandas/general.py [hunk 1]
```diff
@@ -1572,10 +1572,7 @@ def to_datetime(
 
     >>> pd.to_datetime([1, 2, 3], unit='D',
     ...                origin=pd.Timestamp('1960-01-01'))
-    0   1960-01-02
-    1   1960-01-03
-    2   1960-01-04
-    dtype: datetime64[ns]
+    DatetimeIndex(['1960-01-02', '1960-01-03', '1960-01-04'], dtype='datetime64[ns]', freq=None)
```
Comment (sfc-gh-azhan) [inline src/snowflake/snowpark/modin/pandas/general.py:1575]: please also update the type hints (e.g., return type) for this method above.

File: src/snowflake/snowpark/modin/pandas/general.py [hunk 2]
```diff
@@ -2229,9 +2166,11 @@ def date_range(
...
         right_inclusive=right_inclusive,
     )
-    s = Series(query_compiler=qc)
-    s.name = name
-    return s
+    # Set date range as index column.
+    qc = qc.set_index_from_columns(qc.columns.tolist(), include_index=False)
+    # Set index column name.
+    qc = qc.set_index_names([name])
+    return pd.DatetimeIndex(data=qc)
```
Comment (sfc-gh-azhan) [inline src/snowflake/snowpark/modin/pandas/general.py:2173]:
```suggestion
    return pd.DatetimeIndex(query_compiler=qc)
```

File: src/snowflake/snowpark/modin/plugin/extensions/index.py [hunk 3]
```diff
@@ -2353,3 +2354,64 @@ def str(self) -> native_pd.core.strings.accessor.StringMethods:
...
+            yearfirst,
+            utc,
+            format,
+            exact,
+            unit,
+            infer_datetime_format,
+            origin,
+            include_index=True,
+        )
+        return DatetimeIndex(data=new_qc)
```
Comment (sfc-gh-azhan) [inline src/snowflake/snowpark/modin/plugin/extensions/index.py:2417]:
```suggestion
        return DatetimeIndex(query_compiler=new_qc)
```

**Review by sfc-gh-azhan — COMMENTED** [review]
Nice work! Please rebase and make sure all tests pass.

Also can you search `SNOW-1486910` in the whole code path, there are some TODO can be resolved with your change.

**Comment by sfc-gh-nkumar** [comment]
> Nice work! Please rebase and make sure all tests pass.
> 
> Also can you search `SNOW-1486910` in the whole code path, there are some TODO can be resolved with your change.

Done

File: src/snowflake/snowpark/modin/plugin/extensions/index.py [hunk 4]
```diff
@@ -2368,3 +2369,64 @@ def str(self) -> native_pd.core.strings.accessor.StringMethods:
...
+        dayfirst: bool = False,
+        yearfirst: bool = False,
+        utc: bool = False,
+        format: str = None,
+        exact: bool | lib.NoDefault = lib.no_default,
+        unit: str = None,
+        infer_datetime_format: bool | lib.NoDefault = lib.no_default,
+        origin: DateTimeOrigin = "unix",
+    ) -> Index:
+        """
```
Comment (sfc-gh-azhan) [inline src/snowflake/snowpark/modin/plugin/extensions/index.py:2385]: Add one line method description.

**Review by sfc-gh-azhan — APPROVED** [review]
Nice work!

**Pull request merged**